src/main/java/io/neonbee/data/DataVerticle.java

```diff
@@ -130,7 +130,7 @@ public MessageCodec<T, T> getMessageCodec() {
     @Override
     public void init(Vertx vertx, Context context) {
         super.init(vertx, context);
-        JsonObject metrics = getMetricsConfig(NeonBee.get().getConfig().getMetricsConfig());
+        JsonObject metrics = getMetricsConfig(NeonBee.get(vertx).getConfig().getMetricsConfig());
```

Comment:
This line is actually quite important although it has nothing to do with the original change described in the commit message. Maybe it's worth to separate this into a different commit in order to make it clear.

Reply:
Mentioning this as this line bite me today, too when trying to upgrade to the latest vert.x version.

Reply:
I agree. I dropped a line to Benni about it once I saw he updated core and the some of the tests were failing. 

I noticed some regressions due to this while working on some feature in core last week. I discussed with Michael and we agreed to pack it in this commit, thinking this one will be merged soon (I agree it was not the best decision and we should do it in a separate commit).